### PR TITLE
Add Helm chart for deploying to Kubernetes

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/thegreenwebfoundation/grid-intensity-exporter/issues/1

### CODE REVIEW

File: .github/workflows/kubernetes-integration-test.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      -
+        name: Docker build
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          load: true
+          tags: thegreenwebfoundation/grid-intensity-exporter:integration-test
+      - 
+        name: Create kubernetes Kind Cluster
+        uses: helm/kind-action@v1.1.0
```
Comment (rossf7) [inline .github/workflows/kubernetes-integration-test.yaml:28]: The actions creates the [Kind](https://kind.sigs.k8s.io/) cluster which a k8s cluster optimized for testing that runs as a docker container.

File: .github/workflows/kubernetes-integration-test.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - 
+        name: Load image into Kind cluster
+        run: kind load docker-image --name chart-testing thegreenwebfoundation/grid-intensity-exporter:integration-test
```
Comment (rossf7) [inline .github/workflows/kubernetes-integration-test.yaml:31]: This test is similar to the docker test but it loads the image into the kind cluster.

File: .github/workflows/kubernetes-integration-test.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        run: helm install grid-intensity-exporter helm/grid-intensity-exporter --set image.tag=integration-test
+      - 
+        name: Set up Kubectl
+        uses: azure/setup-kubectl@v1
+      - 
+        name: Wait for ready pod
+        run: kubectl wait --for=condition=ready -l app.kubernetes.io/name=grid-intensity-exporter pod
+      - 
+        name: Create port forward to metrics port
+        run: kubectl port-forward deployment/grid-intensity-exporter 8000:8000 &
```
Comment (rossf7) [inline .github/workflows/kubernetes-integration-test.yaml:46]: We need to wait for the pod to be ready and create a port forward connection to access the metrics.

Within the cluster other components can access the exporter via its service.

File: README.md [hunk 4]
```diff
@@ -36,11 +37,29 @@ This provider supports more countries but you will need an API key from
 
 ```sh
 ELECTRICITY_MAP_API_TOKEN=your-api-token
-GRID_INTENSITY_PROVIDER=gridintensity.org.uk
-GRID_INTENSITY_REGION=UK
+GRID_INTENSITY_PROVIDER=electricitymap.org
+GRID_INTENSITY_REGION=IN-KA
```
Comment (rossf7) [inline README.md:41]: Just fixing the example.

File: helm/grid-intensity-exporter/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+apiVersion: "v1"
+description: "A Prometheus exporter for understanding the carbon intensity of compute."
+home: "https://github.com/thegreenwebfoundation/grid-intensity-exporter"
+name: "grid-intensity-exporter"
+version: "0.1.0"
```
Comment (rossf7) [inline helm/grid-intensity-exporter/Chart.yaml:5]: The rest of this is the Helm chart itself. I've tried to keep it simple but as usual there is a lot of YAML 😅

File: .github/workflows/kubernetes-integration-test.yaml (hunk 1 above)
Comment (mrchrisadams) [reply .github/workflows/kubernetes-integration-test.yaml:28]: I didn't know kind existed before this PR. This looks neat!

**Review by mrchrisadams — APPROVED** [review]
I'm happy with this Ross, thanks for taking the time to comment the various parts - I'm not sure I would have understood it all without them 👍

**Pull request merged**